sycl/include/CL/sycl/intel/sub_group_host.hpp

```diff
@@ -23,138 +23,166 @@ struct sub_group {
-    throw runtime_error("Subgroups are not supported on host device. ");
+    throw runtime_error("Subgroups are not supported on host device. ",
+                        PI_INVALID_DEVICE);
   }
 
   template <typename T> T shuffle_down(T x, uint32_t delta) const {
-    throw runtime_error("Subgroups are not supported on host device. ");
+    throw runtime_error("Subgroups are not supported on host device. ",
+                        PI_INVALID_DEVICE);
   }
   template <typename T> T shuffle_up(T x, uint32_t delta) const {
-    throw runtime_error("Subgroups are not supported on host device. ");
+    throw runtime_error("Subgroups are not supported on host device. ",
+                        PI_INVALID_DEVICE);
   }
 
   template <typename T> T shuffle_xor(T x, id<1> value) const {
-    throw runtime_error("Subgroups are not supported on host device. ");
+    throw runtime_error("Subgroups are not supported on host device. ",
+                        PI_INVALID_DEVICE);
   }
 
   /* --- two - input shuffles --- */
   /* indices in [0 , 2* sub - group size ) */
   template <typename T> T shuffle(T x, T y, id<1> local_id) const {
-    throw runtime_error("Subgroups are not supported on host device. ");
+    throw runtime_error("Subgroups are not supported on host device. ",
+                        PI_INVALID_DEVICE);
   }
   template <typename T>
   T shuffle_down(T current, T next, uint32_t delta) const {
-    throw runtime_error("Subgroups are not supported on host device. ");
+    throw runtime_error("Subgroups are not supported on host device. ",
+                        PI_INVALID_DEVICE);
   }
   template <typename T>
   T shuffle_up(T previous, T current, uint32_t delta) const {
-    throw runtime_error("Subgroups are not supported on host device. ");
+    throw runtime_error("Subgroups are not supported on host device. ",
+                        PI_INVALID_DEVICE);
   }
 
   /* --- sub - group load / stores --- */
   /* these can map to SIMD or block read / write hardware where available */
   template <typename T, access::address_space Space>
   T load(const multi_ptr<T, Space> src) const {
-    throw runtime_error("Subgroups are not supported on host device. ");
+    throw runtime_error("Subgroups are not supported on host device. ",
+                        PI_INVALID_DEVICE);
   }
 
   template <int N, typename T, access::address_space Space>
   vec<T, N> load(const multi_ptr<T, Space> src) const {
-    throw runtime_error("Subgroups are not supported on host device. ");
+    throw runtime_error("Subgroups are not supported on host device. ",
+                        PI_INVALID_DEVICE);
   }
 
   template <typename T, access::address_space Space>
   void store(multi_ptr<T, Space> dst, T &x) const {
-    throw runtime_error("Subgroups are not supported on host device. ");
+    throw runtime_error("Subgroups are not supported on host device. ",
+                        PI_INVALID_DEVICE);
```

Comment:
I guess it is more like `feature_not_supported("text", PI_INVALID_OPERATION)`, but I'm okay with this resolution for now